osf-cli/src/main/java/org/dataconservancy/cos/packaging/cli/PackageGenerationApp.java

```diff
@@ -219,12 +198,24 @@ private void run() throws Exception {
                 })
                 .collect(Collectors.toList());
 
+        // Prepare package graph
+        final OsfPackageGraph packageGraph = cxt.getBean("packageGraph", OsfPackageGraph.class);
         packageGraph.add(registration);
         users.forEach(packageGraph::add);
 
-        LinkedHashMap<String, List<String>> metadata = createPackageMetadata();
+        // Prepare content provider using package graph
+        // TODO - Does this work without the lambda-specified resolver used in OsfContentProviderTest?
+        OsfContentProvider contentProvider = new OsfContentProvider(packageGraph);
 
-        Package pkg = ipmPackager.buildPackage(packageGraph, metadata);
+        // Load the metadata file
+        InputStream metadataStream = new FileInputStream(bagMetadataFile);
+
+        // Create the package in the default location with the supplied name.
+        // No package generation parameters are supplied.
+        IpmPackager ipmPackager = new IpmPackager();
+        ipmPackager.setPackageName(packageName);
+        ipmPackager.setPackageLocation(outputLocation.getPath());
+        Package pkg = ipmPackager.buildPackage(contentProvider, metadataStream, null);
```

Comment:
Note that this operation does not currently succeed, as it appears the OSF API changed and our code needs to be updated.